scripts/bin/auto-install

```diff
@@ -73,92 +40,112 @@ rest_call() {
-            exit 1
-        fi
-    done
-
-    install_snap "$_snaps"
+    log "ACKNOWLEDGE RESULT: $_assert acknowledged"
 }
 
 # Install the asserted snap using the snapd REST API
 install_snap() {
-    for _snap in "$@"; do
-        _snap_result="$SNAP_COMMON/${_snap##*/}_result.$_date"
-
-        # install the snap
-        rest_call install "$_snap_result" "$_snap"
-
-        if grep -Iq '"status":"Accepted"' "$_snap_result"; then
-            log "INSTALL RESULT: ${_snap##*/} installed"
-        else
-            log "ERROR: INSTALL RESULT: ${_snap##*/} not installed"
-            cp -f "${SNAP_COMMON}/auto-install.log" "$_mount_point"
-            exit 1
-        fi
+    _snap="$1"
 
-        track_channel "${_snap##*/}" "$_snap_result"
-    done
+    # install the snap
+    _response="$(rest_call install "$_snap")"
+
+    if ! echo "$_response" | grep -Iq '"status":"Accepted"'; then
+        log "ERROR: INSTALL RESULT: ${_snap##*/} not installed"
+        return 1
+    fi
+
+    log "INSTALL RESULT: ${_snap##*/} installed"
 }
 
 # set snapd to explicitly follow stable for a snap
-track_channel() {
-    # keep trying to set the channel to stable because it may take some time if a snap change is in process
-    # give up after 14 loops, where each loop sleeps twice as long as the previous
-    # where 14 loops takes 19.1 minutes
-    _snap_name=${1%%_*}
-    _log_name=$2
+track_stable() {
+    # FIXME:
+    # keep trying to set the channel to stable because it may take some
+    # if a snap change is in proces give up after 14 loops, where
+    # each loop sleeps twice as long as the previou where 14 loops
+    # takes 19.1 minute
+
+    _snap_name="$1"
+
     _loop_count=0
     _sleep=2
 
-    while [ $_loop_count -lt 14 ]; do
-        sleep $_sleep
-        rest_call track "$_log_name" "$_snap_name"
+    while
+        _response="$(rest_call track "$_snap_name")"
 
-        if grep -Iq '"status":"Accepted"' "${_log_name}.channel"; then
+        if echo "$_response" | grep -Iq '"status":"Accepted"'; then
             log "${_snap_name} is now following stable"
-            cp -f "${SNAP_COMMON}/auto-install.log" "$_mount_point"
-            break
+            return
         fi
 
-        _loop_count=$((_loop_count+1))
-        _sleep=$((_sleep*2))
+        [ $_loop_count -lt 9 ]
+    do
+        sleep $_sleep
+
+        : $(( _loop_count += 1 ))
+        : $(( _sleep      *= 2 ))
     done
 
     log "WARN: failed to make $_snap_name follow stable"
     log "WARN: Manual intervention may be required for $_snap_name to refresh"
-    cp -f "${SNAP_COMMON}/auto-install.log" "$_mount_point"
+    return 1
+}
+
+process_mounts() {
+    _watch_file="$1"
+
+    while read -r _mount_point < "$_watch_file"; do
+
+        # Make sure the mount actually exists
+        [ -d "$_mount_point" ] || continue
+
+        # Skip mounts that don't contain any asserts
+        ls "$_mount_point"/* >/dev/null 2>&1 || continue
```

Comment:
Should be `ls "$_mount_point"/*.assert`?
Maybe this should be a `find` instead of an `ls`; supporting arbitrarily nested asserts and snaps is trivial if we use `basename`.

Reply:
Yes lol. You would not believe how many times I have made that mistake.
And yeah, I think using `find` would be a good improvement